File: ProgTechGy/y2xwhj_pt_beadando/src/main/java/adapter_pattern/adapters/PrinterAdapter.java
package adapter_pattern.adapters;

import adapter_pattern.interfaces.Printer;
import adapter_pattern.models.LegacyPrinter;

public class PrinterAdapter implements Printer {
    private LegacyPrinter legacyPrinter;

    public PrinterAdapter(LegacyPrinter legacyPrinter) {
        this.legacyPrinter = legacyPrinter;
    }

    @Override
    public void print(String text) {
        legacyPrinter.printText(text);
    }
}

File: ProgTechEa/MindenEgybe/AdapterPattern/adapters/PrinterAdapter.cs
using AdapterPattern.interfaces;
using AdapterPattern.models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AdapterPattern.adapters
{
    internal class PrinterAdapter : IPrinter
    {
        private LegacyPrinter legacyPrinter;

        public PrinterAdapter(LegacyPrinter legacyPrinter)
        {
            this.legacyPrinter = legacyPrinter;
        }

        public void Print(string content)
        {
            legacyPrinter.LegacyPrint(content);
        }
    }
}
